test/AutoDiff/autodiff_diagnostics.swift

```diff
@@ -170,6 +170,33 @@ struct TF_305 : Differentiable {
   }
 }
 
+// TF-676: Test differentiation of protocol requirement with multiple
+// `@differentiable` attributes.
+protocol MultipleDiffAttrsProto : Differentiable {
+  @differentiable(wrt: (self, x))
+  @differentiable(wrt: x)
+  func f(_ x: Float) -> Float
+}
+func testMultipleDiffAttrsProto<P: MultipleDiffAttrsProto>(_ p: P, _ x: Float) {
+  _ = gradient(at: p, x) { p, x in p.f(x) }
+  _ = gradient(at: x) { x in p.f(x) }
+}
+
+// TF-676: Test differentiation of class method with multiple `@differentiable`
+// attributes.
+class MultipleDiffAttrsClass : Differentiable {
+  @differentiable(wrt: (self, x))
+  @differentiable(wrt: x)
+  func f(_ x: Float) -> Float { x }
+}
+func testMultipleDiffAttrsClass<C: MultipleDiffAttrsClass>(_ c: C, _ x: Float) {
```

Comment:
what is it upcasting to?

Reply:
Upcasting `c` to `MultipleDiffAttrsClass`.
```
[AD] Unhandled instruction in PullbackEmitter:   %5 = upcast %0 : $C to $MultipleDiffAttrsClass  // users: %8, %7, %6
[AD] Diagnosing non-differentiability.
[AD] For instruction:
  %5 = upcast %0 : $C to $MultipleDiffAttrsClass  // users: %8, %7, %6
[AD] With invoker:
(differentiation_invoker autodiff_function_inst=(  %12 = autodiff_function [wrt 0 1] [order 1] %11 : $@callee_guaranteed (@guaranteed C, Float) -> Float // users: %80, %32, %14, %13
))
tf-676.swift:21:26: error: function is not differentiable
  _ = gradient(at: c, x) { c, x in c.f(x) }
                         ^~~~~~~~~~~~~~~~~~
tf-676.swift:21:36: note: expression is not differentiable
  _ = gradient(at: c, x) { c, x in c.f(x) }
                                   ^
```

Reply:
Oh ok. Let's fix that later.